src/main/java/com/amazonaws/encryptionsdk/kmsv2/AwsKmsMrkAwareMasterKey.java

```diff
@@ -151,36 +152,40 @@ public void addGrantToken(final String grantToken) {
     // # master key MUST use the configured AWS KMS client to make an AWS KMS
     // # GenerateDatakey (https://docs.aws.amazon.com/kms/latest/APIReference/
     // # API_GenerateDataKey.html) request constructed as follows:
-    final GenerateDataKeyResult gdkResult =
+    final GenerateDataKeyResponse gdkResponse =
         kmsClient_.generateDataKey(
-            updateUserAgent(
-                new GenerateDataKeyRequest()
-                    .withKeyId(awsKmsIdentifier_)
-                    .withNumberOfBytes(algorithm.getDataKeyLength())
-                    .withEncryptionContext(encryptionContext)
-                    .withGrantTokens(grantTokens_)));
+            GenerateDataKeyRequest.builder()
+                .overrideConfiguration(API_NAME_INTERCEPTOR)
```

Comment:
does the overrideConfig with the API_NAME_INTERCEPTOR do the work that the updateUserAgent method did?

Reply:
yep. that api disappeared in v2. According to the devs, this is the proper way to do it